### PR TITLE
Added pull_request_template.md

### PR DESCRIPTION
**Pull request by danielchen-pyc**
# Changes
- Added pull request template in the `.github` folder. 
- @johnagl 

# Checklist

### CODE REVIEW

File: .github/pull_request_template.md
Comment (jacksonyuanjx) [inline .github/pull_request_template.md]: Should we rename this as "Related Issues" since I don't think we're using tickets in Launch Pad? Please correct me if I'm wrong @johnagl

Comment (danielchen-pyc) [reply .github/pull_request_template.md]: I copied this from the launchpad doc repo's PR template and they were using `tickets` as the title but actually using `closing issues#` in the content. I thought they are the same but I not sure. @johnagl

**Comment by jacksonyuanjx** [comment]
What do you guys think of additional sections such as "How has this been tested?", "Motivation/Context", "Screenshots (if applicable)", etc.?

I found this online and it seems to have some good options too: https://github.com/stevemao/github-issue-templates/blob/master/questions-answers/PULL_REQUEST_TEMPLATE.md

File: .github/pull_request_template.md
Comment (johnagl) [inline .github/pull_request_template.md:5]: It's not too big of a deal, but I guess Issues would be clearer, up to you if you want to change @danielchen-pyc

Comment (johnagl) [reply .github/pull_request_template.md]: Issues is probably a bit clearer, but up to you if you want to change @danielchen-pyc :)

**Comment by danielchen-pyc** [comment]
Since this PR hasn't been merged yet, I changed some features regarding the PR template. Below is the new PR template (and PR also loll). 

# Changes
- Changed tickets to issues
- Added the three sections below for more descriptive explanation of the changes. 

## Motivation and Context



## How Has This Been Tested?




## Screenshots (if appropriate):

# Related Issues
- Closes issue #

# Checklist

**Review by johnagl — APPROVED** [review]
Awesome @danielchen-pyc!

**Review by jacksonyuanjx — APPROVED** [review]

**Comment by danielchen-pyc** [comment]
@johnagl @jacksonyuanjx Hello! I'm still getting familiar with this issue workflow. I was wondering if what I did was correct. So basically, after this PR was approved, I closed issue #7 since PR template has been created and approved. I merge `daniel` branch to `master` and safely deleted the temporary `daniel` branch. Did I do anything wrong? Thank you!!

**Comment by jacksonyuanjx** [comment]
> @johnagl @jacksonyuanjx Hello! I'm still getting familiar with this issue workflow. I was wondering if what I did was correct. So basically, after this PR was approved, I closed issue #7 since PR template has been created and approved. I merge `daniel` branch to `master` and safely deleted the temporary `daniel` branch. Did I do anything wrong? Thank you!!

Hi! What you described sounds good to me, but I'll let John have the final say :)

**Pull request merged**
